test/evergreen/code_complexity_analysis.py

```diff
@@ -0,0 +1,118 @@
+# importing the module
+import argparse
+import json
+import csv
+import os.path
+import ast
+from heapq import nlargest
+import pandas as pd 
+from collections import defaultdict
+    
+class CodeComplexity:
+    dataFrame = pd.DataFrame()
+    topNRegions = 5
+    def __init__(self):
+        self.parser = argparse.ArgumentParser()
+        self.parser.add_argument('-s', '--summary', required=True, help='Path to the complexity summary data file')
+        self.parser.add_argument('-o', '--outfile', help='Path of the file to write test output to')
+        self.parser.add_argument('-d', '--data_file', help='Code complexity data file')
+
+    def run(self):
+        self.args = self.parser.parse_args()
+        self.dataFrame = pd.read_csv(self.args.data_file)
+
+        self.get_atlas_compatible_code_statistics(self.args.summary, self.args.data_file, self.args.outfile)
+
+    # Generate the Atlas compatible format report.
+    def get_atlas_compatible_code_statistics(self, summaryFile, dataFile, outfile='atlas_code_statistics.json'):
+        atlas_format = {
+                    'Test Name': "Code Statistics",
+                    'config': {},
+                    'metrics': self.get_code_statistics(summaryFile, dataFile)
+                }
+
+        dir_name = os.path.dirname(outfile)
+        if dir_name:
+            os.makedirs(dir_name, exist_ok=True)
+
+        with open(outfile, 'w') as outfile:
+            json.dump(atlas_format, outfile, indent=4)
+
+    def get_code_statistics(self, summaryFile, dataFile):
+        resultList = []
+        complexity_avg_dict = {}
+        complexity_avg_dict['name'] = 'Average'
+        complexity_avg_dict['value']=self.get_average(summaryFile)
+
+        complexity_ranges_dict = {}
+        complexity_ranges_dict['name'] = 'Stats Ranges'
+        complexity_ranges_dict['value'] = self.get_complexity_ranges_list(dataFile)
+
+        complexity_regions_dict = {}
+        complexity_regions_dict['name'] = 'Top ' + str(self.topNRegions) + ' Regions'
+        complexity_regions_dict['value'] = self.get_region_list(dataFile)
+
+        resultList.append(complexity_avg_dict)
+        resultList.append(complexity_ranges_dict)
+        resultList.append(complexity_regions_dict)
+
+        return (resultList)
+
+    def get_region_list(self, dataFile):
```

Comment:
Worth a docstring describing we are aimed to collect the top 5 regions/functions with the highest cyclomatic complexity values.

Reply:
Fixed in 7cc186a02197e388c9a2fcaa97331d24c84376a5